packages/lesswrong/lib/ultraFeedSettings.ts

```diff
@@ -0,0 +1,94 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { getBrowserLocalStorage } from '../components/editor/localStorageHandlers';
+import { isClient } from './executionEnvironment';
+
+
+export interface UltraFeedSettings {
+  postTruncationBreakpoints: number[];
+  commentTruncationBreakpoints: number[];
+  collapsedCommentTruncation: number;
+  lineClampNumberOfLines: number;
+}
+
+export const DEFAULT_SETTINGS: UltraFeedSettings = {
+  postTruncationBreakpoints: [100, 500, 1000],
+  commentTruncationBreakpoints: [50, 300, 600],
+  collapsedCommentTruncation: 50,
+  lineClampNumberOfLines: 2,
+};
+
+const ULTRA_FEED_SETTINGS_KEY = 'ultraFeedSettings';
+export const SETTINGS_CHANGE_EVENT = 'ultraFeedSettingsChanged';
+
+export const getUltraFeedSettings = (): UltraFeedSettings => {
+  if (!isClient) return DEFAULT_SETTINGS;
+  
+  const ls = getBrowserLocalStorage();
+  if (!ls) return DEFAULT_SETTINGS;
+  
+  const storedSettings = ls.getItem(ULTRA_FEED_SETTINGS_KEY);
+  if (!storedSettings) return DEFAULT_SETTINGS;
+  
+  try {
+    return { ...DEFAULT_SETTINGS, ...JSON.parse(storedSettings) };
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.error("Failed to parse UltraFeed settings", e);
+    return DEFAULT_SETTINGS;
+  }
+};
+
+export const saveUltraFeedSettings = (settings: Partial<UltraFeedSettings>): void => {
+  if (!isClient) return;
+  
+  const ls = getBrowserLocalStorage();
```

Comment:
It's possible this code would've benefited from `useLocalStorageState`, not sure.

Reply:
Do you think that's where we want to store user settings long term?

Reply:
Shoggoth said that doesn't work for a JSON object